pynicotine/downloads.py

```diff
@@ -531,6 +531,10 @@ def _delete_stale_incomplete_downloads(self):
             if transfer.current_byte_offset and transfer.status != TransferStatus.FINISHED
         }
         incomplete_download_folder_path = self.get_incomplete_download_folder()
```

Comment:
`if self.get_default_download_folder() == incomplete_download_folder_path:` then there is a possibility that the contents of the folder might be huge. In that case, all the shenanigans in this method ought to be avoided entirely.

Reply:
Maybe better to check if we're in the Nicotine+ data folder, and only perform a cleanup in this case? We still want to guarantee that no junk remains in Nicotine+'s "internal" storage, but we'd avoid deleting anything outside of it.

Reply:
That gives different behaviour depending upon what the paths are set to, which makes it difficult to predict what the program will do on any given instance, whereas it shouldn't matter where the incomplete folder is set to.

Perhaps it would be better to delete only those incomplete files which have been created/accessed during a given session, that would involve keeping a transient list of explicit path/filenames which are destined for deletion, rather than just arbitrarily wiping the entire contents of a folder...

... Really, in hindsight an indiscriminate cleanup operation like this ought only be manually triggered by the user. There is also no provision nor possibility of securely erasing the temporary data, which might be something users need.

Reply:
If we can't decide what the best way is, then I suggest either merging what you have so far or removing the feature entirely for now in order to expedite a patch release.

Reply:
i'll merge as-is, we can think more about the feature when there's more time.